dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/Responses/AIAgentResponseExecutor.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Agents.AI.Hosting.OpenAI.Responses.Models;
+using Microsoft.Extensions.AI;
+
+namespace Microsoft.Agents.AI.Hosting.OpenAI.Responses;
+
+/// <summary>
+/// Response executor that uses an AIAgent to execute responses locally.
+/// This is the default implementation for local execution.
+/// </summary>
+internal sealed class AIAgentResponseExecutor : IResponseExecutor
+{
+    private readonly AIAgent _agent;
+
+    public AIAgentResponseExecutor(AIAgent agent)
+    {
+        ArgumentNullException.ThrowIfNull(agent);
+        this._agent = agent;
+    }
+
+    public async IAsyncEnumerable<StreamingResponseEvent> ExecuteAsync(
+        AgentInvocationContext context,
+        CreateResponse request,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        // Create options with properties from the request
+        var chatOptions = new ChatOptions
+        {
+            ConversationId = request.Conversation?.Id,
+            Temperature = (float?)request.Temperature,
+            TopP = (float?)request.TopP,
+            MaxOutputTokens = request.MaxOutputTokens,
+            Instructions = request.Instructions,
+            ModelId = request.Model,
+            AdditionalProperties = new AdditionalPropertiesDictionary
+            {
+                [nameof(CreateResponse)] = request
+            }
+        };
+        var options = new ChatClientAgentRunOptions(chatOptions);
+
+        // Convert input to chat messages
+        var messages = new List<ChatMessage>();
+
+        // If instructions are provided in the request, prepend them as a system message
+        if (!string.IsNullOrWhiteSpace(request.Instructions))
+        {
+            messages.Add(new ChatMessage(ChatRole.System, request.Instructions));
+        }
```

Comment:
For an agent that understands ChatOptions.Instructions, this is going to include the instructions twice. If we're going to include this as a system message, we should probably _not_ also store it as ChatOptions.Instructions.